lib/graphql/object/field.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+# test_via: ../object.rb
+
+module GraphQL
+  class Object < GraphQL::SchemaMember
+    class Field
+      include GraphQL::SchemaMember::CachedGraphQLDefinition
+
+      attr_reader :name
+
+      def initialize(name, return_type_expr = nil, desc = nil, null: nil, field: nil, function: nil, deprecation_reason: nil, method: nil, connection: nil, max_page_size: nil, resolve: nil, &args_block)
```

Comment:
should `connection` be that closely coupled to `GraphQL::Object` or could this be a Relay specialization on this class?

Reply:
Internally, 99% of our connection logic has been extracted to filters anyway, but I think filters are still something that's internal to us? Our `connection` method is literally just two lines now:
- turn `Type` into `TypeConnection`
- apply the pagination filter

Reply:
what's the _benefit_ to extracting it? it's not any easier to maintain, because the reality is that the dependency is very close (that's my experience from the dead `graphql-relay` gem). if anything, I'd like to see the gem become _more_ opinionated on cursor-based pagination, while keeping Relay connections as _one possible_ API for cursor-based pagination. 

or, is there another benefit to separating this code into another file?

Reply:
> I'd like to see the gem become more opinionated on cursor-based pagination, while keeping Relay connections as one possible API for cursor-based pagination.

I'm not sure how specifically baking connections into the gem accomplishes that goal though?

I don't have a strong preference on API here as long as the full implementation is easily pluggable. We have some very shopify-specific things we do to paginate in the backend.

Reply:
Yes, sure, certainly want to _support_ advanced users building their own pagination. But I also want a "pit of success" for new users to get going quickly!

Reply:
I would ❤️ top have a solid API for pagination. At AdHawk we integrate with bunch of APIs which do pagination differently so a mechanism to build connection adapters is incredibly important.